app/assets/v2/js/grants/index.js

```diff
@@ -320,8 +320,13 @@ if (document.getElementById('grants-showcase')) {
         if (!append_mode) {
           vm.grants = [];
         }
+
+        const previouslyLoadedIds = vm.grants.map(grant => grant.id);
+        
         getGrants.grants.forEach(function(item) {
-          vm.grants.push(item);
+          if (!previouslyLoadedIds.includes(item.id)) {
```

Comment:
If `previouslyLoadedIds` was stored as an object against the viewModel we could make this check O(1) instead of O(n) and avoid the need to map `vm.grants` into `previouslyLoadedIds` each call...

```suggestion
          if (!vm.previouslyLoadedIds[item.id]) {
              vm.previouslyLoadedIds[item.id] = item;
```